sw/device/silicon_creator/lib/manifest.h

```diff
@@ -18,6 +18,59 @@
 extern "C" {
 #endif  // __cplusplus
 
+/**
+ * Usage constraints.
+ *
+ * This struct is used to constrain a boot stage image to a set of devices based
+ * on their device IDs, creator and/or owner manufacturing states, and life
+ * cycle states. Bits of `selector_bits` determine which fields (or individual
+ * words of a field as in the case of `device_id`) must be read from the
+ * hardware during verification. Unselected fields must be set to
+ * `MANIFEST_USAGE_CONSTRAINT_UNSELECTED_WORD_VAL` to be able to generate a
+ * consistent value during verification.
+ */
+typedef struct manifest_usage_constraints {
+  /**
+   * Usage constraint selector bits.
+   */
+  uint32_t selector_bits;
+  /**
+   * Device identifier value which is compared against the `DEVICE_ID` value
+   * stored in the `HW_CFG` partition in OTP.
+   *
+   * Mapped to bits 0-8 of `selector_bits`.
+   */
+  uint32_t device_id[8];
```

Comment:
Is the selection granularity sufficient here? IIUC the current scheme means that both the start and end of sub fields in the device ID will need to be 32 bit aligned in order for them to be used as constraints.

Reply:
Correct, selection granularity of this scheme is words but since `device_id` is 8 words long this should give us enough flexibility for different constraints (single device or a family/group of devices, etc.) assuming that `device_id`s are assigned accordingly. That being said, the sizes of the first two fields of the device ID described [here](https://docs.opentitan.org/doc/security/specs/identities_and_root_keys/#device-identifier) are 16-bits. @moidx WDYT?